### ISSUE TITLE
YAML config errors if you have no configuration.

### ISSUE THREAD

**Issue by patricknelson** [silverstripe/silverstripe-framework#5028]
Typically a YAML configuration is expected to have contents, however, due to the framework not first checking to ensure that a `$fragment` is an array, an error will be thrown if you happen to have an empty configuration file (see https://github.com/silverstripe/silverstripe-framework/blob/3/core/manifest/ConfigManifest.php#L655). This should be a quick fix. I think it should be done since sometimes you may need to setup boilerplate code (like in my case) on a per-environment basis separated out into different files but may not have any configuration needed for that particular environment. 

For example, this will cause an error since there's no real data after the second `---`.

```yaml
---
Name: mysite-dev
After:
  - 'framework/*'
  - 'cms/*'
Only:
  environment: dev
---
# Environment: 'dev'
```

**Comment by tractorcow** [comment]
I agree, sounds fair!

The block will still be needed for before/after rules, even if the body is empty.

**Comment by patricknelson** [comment]
@tractorcow  Since I've not gotten to the PR yet -- what branch do you want the PR to be against?

**Comment by tractorcow** [comment]
3 branch is probably safest.

**Comment by patricknelson** [comment]
Ok @tractorcow  see https://github.com/silverstripe/silverstripe-framework/pull/5032

**Comment by tractorcow** [comment]
Thanks for that. ;D Good job.

**Comment by dhensby** [comment]
Fixed by #5032 (i think?)
